src/json-patch-duplex.d.ts

```diff
@@ -5,25 +5,71 @@
  * MIT license
  */
 declare module jsonpatch {
-    function unobserve(root: any, observer: any): void;
-    function observe(obj: any, callback: any): any;
-    function generate(observer: any): any;
+    type Patch = AddPatch | RemovePatch | ReplacePatch | MovePatch | CopyPatch | TestPatch;
+    interface PatchBase {
+        path: string;
+    }
+    interface AddPatch extends PatchBase {
+        op: 'add';
+        value: any;
```

Comment:
Just an idea, how about making `AddPatch` generic? I just recently had an error like this:
```ts
.startWith({ op: 'add', path: '', value: { inIncomplete: true, items: [] } } as AddPatch)
```
which was not caught because `value` is `any`.
`inInComplete` was a typo and should have been `isIncomplete`. That would have been prevented if I could have typed it as `AddPatch<CompletionList>`

Here in the lib code it would just take `Patch<any>` everywhere